File: problems/problem 5/test_cases.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char* encryptString(const char* s);

static int failures = 0;

void check(const char* input, const char* expected) {
    char* output = encryptString(input);
    if (strcmp(output, expected) == 0) {
        printf("PASS\n");
    } else {
        printf("FAIL Input=%s Expected=%s Got=%s\n", input, expected, output);
        failures++;
    }
    free(output);
}

int main() {
    check("hello123", "ihmmr321");
    check("programming2025!", "qsrhsdnnloh5202!"); // fixed
    check("AeiOu", "DhlRx");
    check("testCASE123", "uhtuDDTH321");

    if (failures > 0) return 1;
    return 0;
}

File: problems/problem 5/test_cases.java
public class test_cases {
    private static int failures=0;

    private static void check(String input, String expected) {
        String output = solution.encrypt(input);
        if (output.equals(expected)) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL Input=" + input + " Expected=" + expected + " Got=" + output);
            failures++;
        }
    }

    public static void main(String[] args) {
        check("hello123","igopt321");
        check("programming2025!","qsphsbnnjoh5202!");
        check("AeiOu","DhpRx");
        check("testCASE123","uftuDBTF321");

        if (failures>0) System.exit(1);
        else System.exit(0);
    }
}
